server/WAPI.rb

```diff
@@ -149,6 +152,49 @@ def do_request(request)
     wrapped_response
   end
 
+  ### Log information if a query returns link headers indicating there is more data on another page.
+  ### Currently we only log this information.  We don't go get the additional information.
+
+  def process_link_header(response)
+    ### If there is a 'next' link the headers link section print it along with information for a rough estimate
+    ### of the total number of entries.
+
+    linkheader = LinkHeader.parse(response.headers[:link]).to_a
+
+    next_link, last_link = nil, nil
+
+    #### extract the interesting header links
+    linkheader.each { |link|
+      next_link ||= header_link_for_rel(link, 'next')
+      last_link ||= header_link_for_rel(link, 'last')
+    }
+
+    # If there is more data on another page log that.
+    if !next_link.nil?
+      prefix = ""
+      # Log last_page and per_page values so can get rough estimate of total number of entries for query.
+      if !last_link.nil?
+        # Note: We use the page/per_page information because it appears in the URL, but that isn't a standard
+        # and it might be removed at some point.
+        p = Regexp.new(/page=(\d+)&per_page=(\d+)/)
+        p.match(last_link)
+        last_page, per_page = $1, $2
+        prefix = "last_page: #{last_page} page_size: #{per_page} "
+      end
+      logger.warn "#{self.class.to_s}:#{__method__}:#{__LINE__}: pagination: #{prefix} next_link: #{next_link}"
+    end
+
+  end
+
+  # Extract URL for this link entry if it is the desired link type.
+  def header_link_for_rel(link, desired)
+    link_relationship = link[1][0][1]
```

Comment:
I am confused with the data structure: link[1][0][1], and in the end, why returning link[0]?

More documentation would be appreciated!